mobile-app/app/screens/AppNavigator/AppNavigator.tsx

```diff
@@ -20,6 +27,17 @@ export interface AppParamList {
 export function AppNavigator (): JSX.Element {
   const { isLight } = useThemeContext()
   const DeFiChainTheme: Theme = getDefaultTheme(isLight)
+  const blockCount = useSelector((state: RootState) => state.block.count)
+  const { address } = useWalletContext()
+  const client = useWhaleApiClient()
+  const { network } = useNetworkContext()
+  const dispatch = useDispatch()
+
+  /* Global polling based on blockCount, network and address */
+  useEffect(() => {
```

Comment:
I remembered we have `WalletDataProvider.tsx` for global polling code, to fetch pool pairs and dex price, why not place it there too?

Reply:
@kyleleow It's because of @pierregee PR - https://github.com/DeFiCh/wallet/pull/2718

Reply:
My suggestion is to move `fetchTokens` to `<WalletDataProvider>`, and move `<WalletDataProvider>` to `AppNavigator` from `App.tsx`. 

So we can still keep the global polling code at one file. What do you think?

Reply:
We can. Haven't explored that option yet. What do you think @pierregee ?

Reply:
Yea I agree to keep the maintanability, didn't explore that yet too. I'll have a look!

Reply:
If it doesnt work, i would suggest plan B:
- create another file called `WalletAddressDataProvider` to store global polling code that specifically requires `WalletContext`